File: backend/src/main/java/ma/investpro/entity/AxeAnalytique.java
package ma.investpro.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;

/**
 * Entité représentant un axe analytique
 */
@Entity
@Table(name = "axes_analytiques", uniqueConstraints = {
    @UniqueConstraint(columnNames = "code")
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AxeAnalytique extends BaseEntity {

    @Column(nullable = false, unique = true, length = 50)
    @NotBlank(message = "Le code de l'axe est obligatoire")
    @Size(max = 50, message = "Le code ne peut pas dépasser 50 caractères")
    private String code;

    @Column(nullable = false, length = 200)
    @NotBlank(message = "Le libellé est obligatoire")
    @Size(max = 200, message = "Le libellé ne peut pas dépasser 200 caractères")
    private String libelle;

    @Column(length = 50)
    @Size(max = 50, message = "Le type ne peut pas dépasser 50 caractères")
    private String type; // Département, Centre de coût, Activité, etc.

    @Column(columnDefinition = "TEXT")
    private String description;
}

File: backend/src/main/kotlin/ma/investpro/entity/AxeAnalytique.kt
package ma.investpro.entity

import jakarta.persistence.*
import jakarta.validation.constraints.NotBlank

@Entity
@Table(
    name = "axes_analytiques",
    indexes = [
        Index(name = "idx_axes_analytiques_code", columnList = "code"),
        Index(name = "idx_axes_analytiques_actif", columnList = "actif"),
        Index(name = "idx_axes_analytiques_type", columnList = "type")
    ]
)
class AxeAnalytique(
    @Column(nullable = false, unique = true, length = 50)
    @field:NotBlank
    var code: String = "",

    @Column(nullable = false, length = 200)
    @field:NotBlank
    var libelle: String = "",

    @Column(length = 50)
    var type: String? = null,

    @Column(columnDefinition = "TEXT")
    var description: String? = null

) : BaseEntity()
